### PR TITLE
Ng container shallow query

### LINKED ISSUE #34768: [ivy] ContentChildren doesn't pick up elements inside ng-container
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/core@9.0.0-rc.8

### Is this a regression?
Yes, the previous version in which this bug was not present was: v8

### Description
`ContentChildren` doesn't pick up things marked with `ngProjectAs`.

```html
<app-parent name="indirect with projectAs">
  <ng-container *ngFor="let i of [1, 2, 3, 4]" ngProjectAs="app-child">
    <app-child>
      {{ i }}
    </app-child>
  </ng-container>
</app-parent>
```
```ts
...
export class ParentComponent {
  @ContentChildren(ChildComponent, { descendants: false }) children: QueryList<
    ChildComponent
  >;
}
```
In v8, the `ContentChildren` in the `ParentComponent` would pick up the `app-child` inside the ng-container (technically, it did this regardless of `ngProjectAs`, but the point is it did work). In Ivy, it no longer picks up the children with or without an `ngProjectAs`. Given `<ng-content select=*>` does work with `ngProjectAs`, it seems to me that `ContentChildren` should as well.

## 🔬 Minimal Reproduction
https://github.com/henry-alakazhang/angular-content-children-repro
Master branch is v9, v8 branch is v8.
https://github.com/henry-alakazhang/angular-content-children-repro/tree/v8

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.8
@angular-devkit/build-angular     0.900.0-rc.8
@angular-devkit/build-optimizer   0.900.0-rc.8
@angular-devkit/build-webpack     0.900.0-rc.8
@angular-devkit/core              9.0.0-rc.8
@angular-devkit/schematics        9.0.0-rc.8
@angular/localize                 9.0.0-rc.0
@ngtools/webpack                  9.0.0-rc.8
@schematics/angular               9.0.0-rc.8
@schematics/update                0.900.0-rc.8
rxjs                              6.5.4
typescript                        3.6.4
webpack                           4.41.2
</code></pre>

**Anything else relevant?**
Related to #34289, but the suggested solution (`descendants: true`) is not always feasible. In my case, I want to be able to nest another `parent` inside the `child` and using `descendants: true` would pick up the nested children as well.

Similarly, the other suggestion in [the compatibility guide](https://next.angular.io/guide/ivy-compatibility-examples#content-children-descendants), to only use direct descendants, is difficult to use if I want to apply multiple structural directives to the child, eg.
```html
<app-parent>
  <ng-container *ngFor="...">
    <app-child *ngIf="...">
  </ng-container>
</app-parent>
```

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Fixes #34768

### CODE REVIEW

File: packages/core/test/acceptance/query_spec.ts [hunk 1]
```diff
@@ -715,8 +715,207 @@ describe('query logic', () => {
...
+
+      @Component({selector: 'needs-target', template: ``})
+      class NeedsTarget {
+        @ContentChildren('target') targets !: QueryList<HTMLElement>;
+      }
+      @Component({
+        selector: 'test-cmpt',
+        template: `
+          <needs-target>
+            <ng-container>
```
Comment (mhevery) [inline packages/core/test/acceptance/query_spec.ts]: Could we add a tests when `#target` is on `<ng-container>` As in this case:

```
<needs-target>
  <ng-container #target>
    <b #target></b>
  </ng-container>
</needs-target>
```

It is a bit surprising that we will get both `#target`s

File: packages/core/test/acceptance/query_spec.ts [hunk 2]
```diff
@@ -715,8 +715,207 @@ describe('query logic', () => {
...
+      class NeedsTarget {
+        @ContentChildren(TargetDir) targets !: QueryList<HTMLElement>;
+      }
+
+      @Component({
+        selector: 'test-cmpt',
+        template: `
+          <ng-container targetDir>
+            <ng-container needs-target>
+              <ng-container>
```
Comment (mhevery) [inline packages/core/test/acceptance/query_spec.ts]: ohh, I see this is covered sort of here. (Not sure if previous comment) applies.

**Review by mhevery — APPROVED** [review]

**Comment by mhevery** [comment]
Adding `cleanup` to consider adding additional test, LGTM otherwise.

File: packages/core/test/acceptance/query_spec.ts (hunk 1 above)
Comment (pkozlowski-opensource) [reply packages/core/test/acceptance/query_spec.ts]: yeh, I believe that we've got such a test already, as you've noticed in [another comment](https://github.com/angular/angular/pull/35384#discussion_r379006526).

**Review by kara — CHANGES_REQUESTED** [review]
Mostly ready to go, but can we also add more context to the commit message about what this change is and why?

File: packages/core/test/acceptance/query_spec.ts [hunk 3]
```diff
@@ -715,8 +715,207 @@ describe('query logic', () => {
...
+          class TestCmpt {
+          }
+
+          TestBed.configureTestingModule({declarations: [TestCmpt, NeedsTarget, TargetDir]});
+          const fixture = TestBed.createComponent(TestCmpt);
+          const cmptWithQuery = fixture.debugElement.children[0].injector.get(NeedsTarget);
+
+          fixture.detectChanges();
+          expect(cmptWithQuery.targets.length).toBe(3);
+        });
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply packages/core/test/acceptance/query_spec.ts]: OMG, @kara I didn't have this test and... you know what... there was an actual bug in my impl! Good catch, thank you!

File: aio/scripts/_payload-limits.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
     "master": {
       "uncompressed": {
         "runtime-es2015": 2987,
-        "main-es2015": 448928,
+        "main-es2015": 449449,
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply aio/scripts/_payload-limits.json]: yep,  size accumulating over commits... I've rebased this PR and now I don't have to bump those limits any more...

File: packages/core/test/acceptance/query_spec.ts [hunk 5]
```diff
@@ -715,8 +715,207 @@ describe('query logic', () => {
...
+      })
+      class TestCmpt {
+      }
+
+      TestBed.configureTestingModule({declarations: [TestCmpt, NeedsTarget]});
+      const fixture = TestBed.createComponent(TestCmpt);
+      const cmptWithQuery = fixture.debugElement.children[0].injector.get(NeedsTarget);
+
+      fixture.detectChanges();
+      expect(cmptWithQuery.targets.length).toBe(1);
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply packages/core/test/acceptance/query_spec.ts]: Added asserts for query result type.

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
[presubmit](http://test/OCL:295769176:BASE:295769224:1582051439482:5364afc6)

**Pull request closed without merging**